backend-lib/src/main/rust/lib.rs

```diff
@@ -589,10 +578,10 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_is
         let network = parse_network(network_id as u32)?;
         let addr = utils::java_string_to_rust(&env, addr);
 
-        match RecipientAddress::decode(&network, &addr) {
+        match Address::decode(&network, &addr) {
             Some(addr) => match addr {
-                RecipientAddress::Shielded(_) => Ok(JNI_TRUE),
-                RecipientAddress::Transparent(_) | RecipientAddress::Unified(_) => Ok(JNI_FALSE),
+                Address::Sapling(_) => Ok(JNI_TRUE),
+                Address::Transparent(_) | Address::Unified(_) => Ok(JNI_FALSE),
```

Comment:
nit: although this is not a change to behavior from this PR, and is instead just highlighted because of the rename, I'm now confused about what the behavior here is intended to be.

Does the name of the `isValidShieldedAddress` method need to change to `isValidSaplingAddress`? I don't understand why a Unified address is not considered a valid shielded address, given that it must have a shielded receiver.

Reply:
Yeah, the FFI method should be renamed as well.

Reply:
Hm, this is actually exposed all the way to the outer SDK API. I'm only going to rename the FFI layer and then the SDK part can be decided later.